### DISCUSSION TITLE
using the commands to register devices and download profiles

### DISCUSSION THREAD

**Discussion by moayyad3** [fastlane/fastlane#17494] [Q&A] [answered]
Can someone help me please setting up the commands or setting up a script about auto registering the devices using the api and downloading profile .

**Comment by joshdholtz** [comment, marked as answer]
## With a Fastfile do 

Put something like this in your `Fastfile`...

```rb
lane :add_device do
  register_device(
    name: "Someone’s iPhone 6", # Optional (will prompt during run)
    udid: "1234567890123456789012345678901234567890", # Optional (will prompt during run)
    team_id: "XXXXXXXXXX",         # Optional, if you"re a member of multiple teams, then you need to pass the team ID here.
    username: "your@email.com"   # Optional, lets you override the Apple Member Center username.
  )

  sigh(
    development: true, # Use either this line or the one below
    adhoc: true, # Or this one (not sure if you are doing development or adhoc)
    force: true # Needed for when new devices added
  )
end
```

## You can also do the same with just CLI

```sh
fastlane run register_device --name “Something” --udid “Something” --team_id “Something” --username “Something”
fastlane sigh --force
```

sigh docs - http://docs.fastlane.tools/actions/sigh/#sigh
register_device docs - http://docs.fastlane.tools/actions/register_device/#register_device

**Reply by moayyad3** [reply]
how can I make it ask for the udid and name everytime i use the command instead of manually editing the file ?

and where do I put the API keys ?

**Reply by rogerluan** [reply]
As the code comments say, the name and the udid are optional and if you don't provide them, you will be prompted to enter them when you run the command 😉 @moayyad3
